electron/launcher.ts

```diff
@@ -485,20 +485,22 @@ async function runLegendaryOrGogdlCommand(
 ): Promise<ExecResult> {
   const fullRunnerPath = join(runner.dir, runner.bin)
   const appName = commandParts[commandParts.findIndex(() => 'launch') + 1]
+
+  // Necessary to get rid of undefined or null entries, else
+  // TypeError is triggered
+  commandParts = commandParts.filter(Boolean)
   const safeCommand = getLegendaryOrGogdlCommand(
     commandParts,
     options?.env,
     options?.wrappers,
     fullRunnerPath
   )
-  logDebug(['Running', runner.name, 'command:', safeCommand], runner.logPrefix)
-  if (options?.logFile) {
-    logDebug([`Logging to file "${options.logFile}"`], runner.logPrefix)
-  }
 
-  commandParts = commandParts.filter((n) => n)
```

Comment:
Removing this introduced the TypeErrors mentioned above

Reply:
I know. I putted in runLegendaryOrGodlCommand. I can also put it here. Imported was the comment, else it is to confusing.

Reply:
My main reason for having this check wasn't to prevent undefined/null anyways. I wanted to filter out empty arguments (which I guess can't happen anymore, but they added unnecessary spaces to the launch command back then)
Again, if you pass something other than the annotated type to other functions, they would most likely also break. A function indicating that `string[]` is the expected type should not have to worry about being given `undefined`/`null`, otherwise we'd have 100s of lines of boilerplate checking for this everywhere

Reply:
This happens because sometimes variables joined together for the commandParts are sometimes undefined. The thing for me here was that this line is unclear. It looks like unessecary. Something like this is much more readable:
`commandParts = commandParts.filter((part) => {if(part) {return part}}`